lib/internal/Magento/Framework/Url.php

```diff
@@ -457,8 +457,13 @@ public function getBaseUrl($params = [])
          */
         $origScope = $this->_getScope();
 
-        if (isset($params['_scope'])) {
-            $this->setScope($params['_scope']);
+        /**
+         * Custom scope
+         */
+        $customScope = $this->getRouteParamsResolver()->getRouteParam('_scope');
+
+        if (isset($params['_scope']) || $customScope) {
+            isset($params['scope']) ? $this->setScope($params['_scope']) : $this->setScope($customScope);
```

Comment:
Operating `$customScope`, `$params['scope']` and `$params['_scope']` is quite confusing. What do you think about extracting the scope setting login to a private method with a meaningful description and parameter names to make this code easier to understand?